### PR TITLE
configure circleci

### PR DESCRIPTION
**Pull request by sunchunqiang**
implement 
build
unit test
release tag
release to maven 
release javadoc

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -24,6 +24,63 @@ allprojects {
...
+
+
+
+
+    subprojects.each { proj ->
+        proj.evaluate()
+        proj.tasks.withType(Javadoc).each { javadocTask ->
+            source += javadocTask.source
+            classpath += javadocTask.classpath
+            excludes += javadocTask.excludes
```
Comment (minbi) [inline build.gradle]: Can you exclude all internal packages

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            bash gradlew generateJavadocs
+            git checkout -- gradle.properties 
+
+      - run:
+          name: copy documents
+          command: |
+            git config --local user.name "${CIRCLE_PROJECT_USERNAME}"
+            git checkout  gh-pages
+            cp -R build/docs/javadoc/*   docs/reference/
+
```
Comment (minbi) [inline .circleci/config.yml]: You should preserve this path `docs/2.7.7/reference`

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+      - run:
+          name: checkin documents
```
Comment (minbi) [inline .circleci/config.yml]:
```suggestion
          name: check in documents
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+          command: |
+            git add docs/reference
+            git commit -m "Update document via CircleCI"
```
Comment (minbi) [inline .circleci/config.yml]: this should be the SDK version
AWS SDK for Android 2.8.2

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+            git push -q https://${GH_TOKEN}@github.com/aws-amplify/aws-sdk-android.git
```
Comment (minbi) [inline .circleci/config.yml]: why do we need quiet here?

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+            echo 'export ANDROID_BUILDTOOL_VERSION="27.0.1"' >> $BASH_ENV
+
+            echo "build-tools;$ANDROID_BUILDTOOL_VERSION" 
+
+            sdkmanager "platforms;android-21" "build-tools;27.0.1"  "extras;google;m2repository" "extras;android;m2repository"          
+                
+      - run:
+          name: download gpghome
+          command: |
+            aws s3 cp s3://gpghome-csun/gpghome.zip gpghome.zip
```
Comment (minbi) [inline .circleci/config.yml]: This should be the team's credentials

File: .circleci/config.yml [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+            unzip -a gpghome.zip
+            echo "gpghome contents:"
+            ls gpghome
+
+      - run:
+         name: install maven
+         command: |
+          sudo apt-get install maven
+      - run:
```
Comment (minbi) [inline .circleci/config.yml]: This is not needed

File: aws-android-sdk-autoscaling/build.gradle [hunk 8]
```diff
@@ -8,3 +8,8 @@ dependencies {
 
 sourceCompatibility = "1.7"
 targetCompatibility = "1.7"
+
+task myJavadocs11(type: Javadoc) {
```
Comment (minbi) [inline aws-android-sdk-autoscaling/build.gradle]: Does this need to be here?

File: aws-android-sdk-kinesisvideo/pom.xml [hunk 9]
```diff
@@ -57,7 +57,7 @@
        <artifactId>android</artifactId>
        <version>5.0</version>
        <scope>system</scope>
-       <systemPath>${basedir}/../android-21-linux.jar</systemPath>
```
Comment (minbi) [inline aws-android-sdk-kinesisvideo/pom.xml]: These changes will be reverted by the release process.

**Comment by sunchunqiang** [comment]
create another pr for the issue

**Pull request closed without merging**